ampd/src/handlers/config.rs

```diff
@@ -0,0 +1,84 @@
+use serde::de::{self, Deserializer};
+use serde::Deserialize;
+use std::collections::HashSet;
+
+use crate::evm::ChainName;
+use crate::types::TMAddress;
+use crate::url::Url;
+
+#[derive(Debug, Deserialize)]
+pub struct Chain {
+    pub name: ChainName,
+    pub rpc_url: Url,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(tag = "type")]
+pub enum Config {
+    EVMMsgVerifier {
+        chain: Chain,
+        cosmwasm_contract: TMAddress,
+    },
+    MultisigSigner {
+        cosmwasm_contract: TMAddress,
+    },
+}
+
+fn validate_multisig_signer_config<'de, D>(configs: &[Config]) -> Result<(), D::Error>
+where
+    D: Deserializer<'de>,
+{
+    match configs
+        .iter()
+        .filter(|config| matches!(config, Config::MultisigSigner { .. }))
+        .count()
+    {
+        count if count > 1 => Err(de::Error::custom(
+            "only one multisig signer config is allowed",
+        )),
+        _ => Ok(()),
+    }
+}
+
+fn validate_evm_msg_verifier_configs<'de, D>(configs: &[Config]) -> Result<(), D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let evm_msg_verifier_configs: Vec<_> = configs
+        .iter()
+        .filter(|config| matches!(config, Config::EVMMsgVerifier { .. }))
+        .collect();
```

Comment:
filter_map on chain here already? Then you just need to convert it into a HashSet in the next step

Reply:
hmm let me try if I can make it clearer. The filter part of the `filter_map` below isn't really doing anything.